### PR TITLE
Added test for interpolate nearest exact

### PR DESCRIPTION
**Pull request by Anstow**

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -4215,6 +4215,7 @@ def sample_inputs_interpolate(mode, self, device, dtype, requires_grad, **kwargs
         align_corners_options = (True, False, None)
     ranks_for_mode = {
         'nearest': [1, 2, 3],
+        'nearest_exact': [1, 2, 3],
```
Comment (mikaylagawarecki) [inline torch/testing/_internal/common_methods_invocations.py:4218]: seems like this should be 
```suggestion
        'nearest-exact': [1, 2, 3],
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -12824,6 +12825,22 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
                DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
            ),
            supports_out=False),
+    OpInfo('nn.functional.interpolate',
+           aten_name="interpolate",
+           variant_test_name='nearest_exact',
```
Comment (mikaylagawarecki) [inline torch/testing/_internal/common_methods_invocations.py:12830]:
```suggestion
           variant_test_name='nearest-exact',
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -12824,6 +12825,22 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
...
+           supports_autograd=True,
+           supports_fwgrad_bwgrad=True,
+           supports_forward_ad=True,
+           dtypes=floating_types_and(torch.uint8, torch.bfloat16),
+           dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16, torch.uint8),
+           sample_inputs_func=partial(sample_inputs_interpolate, 'nearest_exact'),
```
Comment (mikaylagawarecki) [inline torch/testing/_internal/common_methods_invocations.py:12836]:
```suggestion
           sample_inputs_func=partial(sample_inputs_interpolate, 'nearest-exact'),
```

**Review by mikaylagawarecki — COMMENTED** [review]
Thanks, seems like there is a small typo in the mode

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -12824,6 +12825,31 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
...
+           sample_inputs_func=partial(sample_inputs_interpolate, 'nearest-exact'),
+           skips=(
+               # RuntimeError: false
+               # INTERNAL ASSERT FAILED at "../torch/csrc/jit/passes/utils/check_alias_annotation.cpp":185,
+               # please report a bug to PyTorch.
+               DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
+               # RuntimeError: aten::_upsample_nearest_exact*d hit the vmap fallback which is currently disabled
+               DecorateInfo(unittest.expectedFailure, 'TestOperators', 'test_vmapjvpall_has_batch_rule'),
+               DecorateInfo(unittest.expectedFailure, 'TestOperators', 'test_vmapvjp_has_batch_rule'),
+               DecorateInfo(unittest.expectedFailure, 'TestVmapOperatorsOpInfo', 'test_vmap_exhaustive'),
```
Comment (mikaylagawarecki) [inline torch/testing/_internal/common_methods_invocations.py:12845]: Seems like this is yielding an unexpected success (see [here ](https://github.com/pytorch/pytorch/actions/runs/6122316764/job/16618236122))

**Review by mikaylagawarecki — COMMENTED** [review]
The failing CI is because `interpolate_nearest_exact` does not have a vmap batching rule. You could add an `xfail` for it over here

https://github.com/pytorch/pytorch/blob/39ff80125f5c6d240a4b59010253cb4adab5090f/test/functorch/test_vmap.py#L3756-L3758

and perhaps file an issue to track this

**Comment by Anstow** [comment]
> You could add an `xfail` for it over here

Sometimes it looks like the expected failures are on the op and some of them are on the test. Is there any logic behind this? I've added it on the op because I did it before reading your comment. Is it worth moving the expected failure?

File: torch/testing/_internal/common_methods_invocations.py (hunk 4 above)
Comment (Anstow) [reply torch/testing/_internal/common_methods_invocations.py]: It's now passing on cuda too... I guess they might have fixed it

**Review by mikaylagawarecki — APPROVED** [review]
Thanks!

**Comment by mikaylagawarecki** [comment]
Seems like there is one more test that needs to be skipped on MPS https://hud.pytorch.org/pr/108558

**Comment by Anstow** [comment]
@mikaylagawarecki  For future reference can I merge using the bot myself now it's approved or should I wait for the reviewer to do that?

**Comment by mikaylagawarecki** [comment]
@Anstow in future, feel free to merge yourself you can type `@pytorchbot --help` for instructions

**Pull request closed without merging**